src/screens/AddNotificationEmailScreen.tsx

```diff
@@ -41,21 +48,63 @@ class AddNotificationEmailScreen extends PureComponent<Props, State> {
     });
   };
 
-  onSave = () => {
+  goToLocalEmailConfirm = () => {
+    const { setNotificationEmail, navigation, createNotificationEmail, route } = this.props;
+
+    const { onSuccess, title } = route.params;
+
     const { email } = this.state;
-    const { setNotificationEmail, navigation, hasWallets } = this.props;
+
+    setNotificationEmail(email, {
+      onSuccess: () =>
+        navigation.navigate(Route.LocalConfirmNotificationCode, {
+          children: (
+            <View style={styles.infoContainer}>
+              <Text style={typography.headline4}>{i18n.onboarding.confirmEmail}</Text>
+              <Text style={styles.codeDescription}>{i18n.onboarding.confirmEmailDescription}</Text>
+              <Text style={typography.headline5}>{email}</Text>
+            </View>
+          ),
+          title,
+          onSuccess: () => {
+            createNotificationEmail(email, {
+              onSuccess,
+            });
+          },
+        }),
+    });
+  };
+
+  onConfirm = () => {
+    const { email } = this.state;
+    const { navigation, hasWallets, route, checkSubscription, wallets } = this.props;
+    const { onSuccess } = route.params;
+
     if (!isEmail(email)) {
       return this.setState({
         error: i18n.onboarding.emailValidation,
       });
     }
     if (!hasWallets) {
-      setNotificationEmail(email, {
-        onSuccess: () => navigation.navigate(Route.ConfirmNotificationCode, { email }),
-      });
-    } else {
-      navigation.navigate(Route.ChooseWalletsForNotification, { email, isOnboarding: true });
+      return this.goToLocalEmailConfirm();
     }
+    checkSubscription(wallets, email, {
+      onSuccess: (ids: string[]) => {
+        const walletsToSubscribe = wallets.filter(w => !ids.includes(w.id));
```

Comment:
we pass only wallets to subscribe, not all wallets as before because it's possible that the user has subscribed wallets on different devices. The ChooseWalletsForNotificationScreen is responsible only for choosing wallets, I pass it forward (currently to ConfirmEmail) when they are hashed, etc.